src/elasticarray.jl

```diff
@@ -57,76 +57,132 @@ ElasticArray(A::AbstractArray{T,N}) where {T,N} = copyto!(ElasticArray{T,N}(unde
-import Base.==
-(==)(A::ElasticArray, B::ElasticArray) =
-    ndims(A) == ndims(B) && A.kernel_size == B.kernel_size && A.data == B.data
+@inline Base.size(A::ElasticArray) = (A.kernel_size..., div(length(eachindex(A.data)), A.kernel_length))
 
+@inline Base.length(A::ElasticArray) = length(A.data)
 
-Base.parent(A::ElasticArray) = A.data
+@propagate_inbounds Base.getindex(A::ElasticArray, i::Int) = getindex(A.data, i)
 
-Base.size(A::ElasticArray) = (A.kernel_size..., div(length(eachindex(A.data)), A.kernel_length))
-@propagate_inbounds Base.getindex(A::ElasticArray, i::Integer) = getindex(A.data, i)
-@propagate_inbounds Base.setindex!(A::ElasticArray, x, i::Integer) = setindex!(A.data, x, i)
-@inline Base.IndexStyle(A::ElasticArray) = IndexStyle(A.data)
+@propagate_inbounds Base.setindex!(A::ElasticArray, x, i::Int) = setindex!(A.data, x, i)
 
-Base.length(A::ElasticArray) = length(A.data)
+Base.IndexStyle(::Type{<:ElasticArray}) = IndexLinear()
 
-Base.dataids(A::ElasticArray) = Base.dataids(A.data)
 
 
-@inline function Base.resize!(A::ElasticArray{T,N}, dims::Vararg{Integer,N}) where {T,N}
-    kernel_size, size_lastdim = _split_resize_dims(A, dims)
+@inline Base.resize!(A::ElasticArray{T,N}, dims::Vararg{Integer,N}) where {T,N} = resize!(A, dims)
+
+@inline function Base.resize!(A::ElasticArray{T,N}, dims::NTuple{N,Integer}) where {T,N}
+    _, size_lastdim = _split_resize_dims(A, dims)
     resize!(A.data, A.kernel_length.divisor * size_lastdim)
-    A
+    return A
+end
+
+@inline function Base.resize!(A::ElasticArray, size_lastdim::Integer)
+    resize!(A.data, A.kernel_length.divisor * size_lastdim)
+    return A
+end
+
+
+@inline Base.sizehint!(A::ElasticArray{T,N}, dims::Vararg{Integer,N}) where {T,N} = sizehint!(A, dims)
+
+@inline function Base.sizehint!(A::ElasticArray{T,N}, dims::NTuple{N,Integer}) where {T,N}
+    _, size_lastdim = _split_resize_dims(A, dims)
+    sizehint!(A.data, A.kernel_length.divisor * size_lastdim)
+    return A
 end
 
-@inline function Base.sizehint!(A::ElasticArray{T,N}, dims::Vararg{Integer,N}) where {T,N}
-    kernel_size, size_lastdim = _split_resize_dims(A, dims)
+@inline function Base.sizehint!(A::ElasticArray, size_lastdim::Integer)
     sizehint!(A.data, A.kernel_length.divisor * size_lastdim)
-    A
+    return A
 end
 
+
 function _split_resize_dims(A::ElasticArray, dims::NTuple{N,Integer}) where {N}
     kernel_size, size_lastdim = _split_dims(dims)
     kernel_size != A.kernel_size && throw(ArgumentError("Can only resize last dimension of an ElasticArray"))
-    kernel_size, size_lastdim
+    return kernel_size, size_lastdim
 end
 
 
+
 function Base.append!(dest::ElasticArray, src::AbstractArray)
     rem(length(eachindex(src)), dest.kernel_length) != 0 && throw(DimensionMismatch("Can't append, length of source array is incompatible"))
     append!(dest.data, src)
-    dest
+    return dest
 end
 
+function Base.append!(dest::ElasticArray, iter)
+    for el in iter
+        append!(dest, el)
```

Comment:
I see your point. It would still be nice to support appending/prepending from any iterable to be consistent with `Base` (e.g. `append!(ElasticArray{Float64}(undef, 2, 3), (rand() for _=1:2))`). If we changed this to: 
```julia
function Base.append!(dest::ElasticArray, iter)
    append!(dest.data, iter)
    # check rem(length(eachindex(A.data)), A.kernel_length) == 0
    return dest
end
 ```
then it would be consistent with the semantics of `append!`, correct? Since we don't have access to `length` for an arbitrary iterator we'd have to check if `rem(length(eachindex(A.data)), A.kernel_length) == 0` after appending/prepending.

Reply:
Sounds good to  me!

Reply:
See https://github.com/JuliaArrays/ElasticArrays.jl/pull/20/commits/fc1f4e7d8fa9b33c6df664596e65086c8df1b2fe. Turns out we don't need a separate `prepend!`/`append!` when `src` is an `AbstractArray` :).